tests/test_new_interface.py

```diff
@@ -0,0 +1,66 @@
+from impy.constants import TeV
+from impy.kinematics import EventKinematics
+import impy.models as im
+from impy.common import MCRun
+from collections import Counter
+import pytest
+from multiprocessing import Pool
+from multiprocessing.context import TimeoutError
+
+
+models = set()
+for key in dir(im):
+    obj = getattr(im, key)
+    if hasattr(obj, "__mro__") and MCRun in obj.__mro__:
+        models.add(obj)
+
+
+def run_model(model, ekin):
+    gen = model(ekin)
+
+    c = Counter()
+    for event in gen(10):
+        event.filter_final_state()
+        assert len(event.p_ids) > 0
+        c.update(event.p_ids)
+
+    return c
+
+
+@pytest.mark.parametrize("model", models)
+def test_new_interface(model):
+
+    p1pdg = -211  # pi-
+    p2pdg = 2212  # proton
+    if model is im.Sophia20:
+        # Sophia can only do γp, γn
+        p1pdg = 22  # gamma
+    elif model in [im.Phojet112, im.UrQMD34]:
+        # The old phojet needs more tweaking for pion-proton (is not related to test)
+        p1pdg = 2212  # proton
+    # elif model is models.Pythia6:
+    #     # Pythia6 can only do ee, ep, pp
+    #     p1pdg = 2212  # proton
+
+    ekin = EventKinematics(
+        ecm=7 * TeV,
+        p1pdg=p1pdg,
+        p2pdg=p2pdg,
+    )
+
+    # Some models need to initialize same fortran code,
+    # which can only be initialized once, therefore run
+    # in separate thread
+    # maxtasksperchild=1 is needed to prevent multiproc to recycle workers and take for each task a fresh interpreter.
+    with Pool(1, maxtasksperchild=1) as p:
```

Comment:
This is not necessary, so I am inclined to revert this.
1) The pool only exists to create one job and then destructs. So there is only ever one task executed per child.
2) It was working fine without.

Reply:
Right. That's different in how Pool is used test_all_cms. I'm fine with reverting..

Reply:
One reason to leave it is as a reminder that  maxtasksperchild=1 is necessary in general for Pool, to prevent copy/paste errors from this test in the future.

Reply:
Ok, let me just add that as a comment and then leave it as is